rust/lance/src/dataset/scanner.rs

```diff
@@ -522,15 +573,101 @@ impl Scanner {
             extra_columns.push(vector_field);
             extra_columns.push(ArrowField::new(DIST_COL, DataType::Float32, true));
         };
+
         if self.with_row_id {
             extra_columns.push(ROW_ID_FIELD.clone());
         }
 
         let schema = if !extra_columns.is_empty() {
-            self.projections.merge(&ArrowSchema::new(extra_columns))?
+            self.phyical_columns
+                .merge(&ArrowSchema::new(extra_columns.clone()))?
         } else {
-            self.projections.clone()
+            self.phyical_columns.clone()
         };
+
+        // drop metadata
+        // NOTE: this is the current behavior as we don't return metadata in queries
+        // We should reconsider this behavior
+        if self.nearest.is_some() {
+            Ok(Arc::new(Schema {
+                fields: schema.fields,
+                metadata: HashMap::new(),
+            }))
+        } else {
+            Ok(Arc::new(schema))
+        }
+    }
+
+    pub(crate) fn output_expr(&self) -> Result<Vec<(Arc<dyn PhysicalExpr>, String)>> {
+        let physical_schema = self.physical_schema()?;
```

Comment:
Can't we just take the schema off of the parent exec node? That feels more composable. And then you shouldn't need to modify the semantics of physical schema.

Reply:
I did hink about that. I erred on the side of a more conservative change as we need to resolve this before `create_plan` to return `output_schema`. Let me do this. we can just change `output_schema = create_plane()?.schema()`

Reply:
Ah I see. having to make the schema available before we've actually built the plan is PINA. let's keep as-is for now, and we can fix this later.